woocommerce/woocommerce
# Remove foreign key from the wc_download_log table

### All Submissions:






### Changes proposed in this Pull Request:



Closes #34305  .

This is the only use of FK in our whole db schema and seems unnecessary, as it forces the use of InnoDB engine. We don't use FKs elsewhere, so it's better to remove it here as well.


### How to test the changes in this Pull Request:

1.  Run the upgrade and make sure the foreign key was removed.
2. Create a downloadable product
3. Buy it a couple of times with different customers and download the file(s) to create download log entries.
4. Check the permission table (`wp_woocommerce_downloadable_product_permissions`) and download log table (`wp_wc_download_log`) to make sure there are some records 
5. Try to programmatically remove the download log through all the data stores/abstractions above.
6. Set some permissions as expired (change the date to yesterday) in the database (e.g. via phpMyAdmin)
7. Go to WC > Settings > Tools and run the tool that removes expired access rights to files.
8. Make sure they were removed.

### Other information:




### FOR PR REVIEWER ONLY:
